src/main/host/syscall_condition.c

```diff
@@ -310,6 +309,8 @@ static void _syscallcondition_signal(Host* host, void* obj, void* arg) {
     bool wasTimeout = (bool)arg;
     MAGIC_ASSERT(cond);
 
+    cond->signalPending = false;
```

Comment:
Might we worth a comment:
The signal task is now being executed by the worker and is no longer pending in the host's event queue. This is necessary, since it's possible that we want to schedule another signal task before this condition object is destroyed.

Reply:
Done